src/analysis/processing/qgsalgorithmdelaunaytriangulation.cpp

```diff
@@ -0,0 +1,174 @@
+}
+
+QString QgsDelaunayTriangulationAlgorithm::displayName() const
+{
+  return QObject::tr( "Delaunay triangulation" );
+}
+
+QStringList QgsDelaunayTriangulationAlgorithm::tags() const
+{
+  return QObject::tr( "delaunay,triangulation,polygons,voronoi" ).split( ',' );
+}
+
+QString QgsDelaunayTriangulationAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsDelaunayTriangulationAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsDelaunayTriangulationAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Creates a polygon layer with the Delaunay triangulation corresponding to a points layer." );
+}
+
+QgsDelaunayTriangulationAlgorithm *QgsDelaunayTriangulationAlgorithm::createInstance() const
+{
+  return new QgsDelaunayTriangulationAlgorithm();
+}
+
+void QgsDelaunayTriangulationAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  std::unique_ptr<QgsProcessingParameterNumber> toleranceParam = std::make_unique<QgsProcessingParameterNumber>( QStringLiteral( "TOLERANCE" ), QObject::tr( "Tolerance" ), QgsProcessingParameterNumber::Double, 0, true, 0 );
+  toleranceParam->setHelp( QObject::tr( "Specifies an optional snapping tolerance which can be used to improve the robustness of the triangulation" ) );
+  addParameter( toleranceParam.release() );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "ADD_ATTRIBUTES" ), QObject::tr( "Add point IDs to output" ), true ) );
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Delaunay triangulation" ), QgsProcessing::TypeVectorPolygon ) );
+}
+
+QVariantMap QgsDelaunayTriangulationAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  std::unique_ptr< QgsProcessingFeatureSource > source( parameterAsSource( parameters, QStringLiteral( "INPUT" ), context ) );
+  if ( !source )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "INPUT" ) ) );
+
+  if ( source->featureCount() < 3 )
+    throw QgsProcessingException( QObject::tr( "Input layer should contain at least 3 points." ) );
+
+  const double tolerance = parameterAsDouble( parameters, QStringLiteral( "TOLERANCE" ), context );
+  const bool addAttributes = parameterAsBool( parameters, QStringLiteral( "ADD_ATTRIBUTES" ), context );
+
+  QgsFields fields;
+  if ( addAttributes )
+  {
+    fields.append( QgsField( QStringLiteral( "POINTA" ), QVariant::LongLong ) );
+    fields.append( QgsField( QStringLiteral( "POINTB" ), QVariant::LongLong ) );
+    fields.append( QgsField( QStringLiteral( "POINTC" ), QVariant::LongLong ) );
+  }
+  else
+  {
+    fields.append( QgsField( QStringLiteral( "id" ), QVariant::LongLong ) );
+  }
```

Comment:
@alexbruy I'm on my way to document this PR and wonder if it is deliberate that the output layer contains either an id field or the IDs from the input point. Why not always have the generated id field in the output layer (and add or not the original points)? I guess, something like:
```
  QgsFields fields;
  fields.append( QgsField( QStringLiteral( "id" ), QVariant::LongLong ) );
  if ( addAttributes )
  {
    fields.append( QgsField( QStringLiteral( "POINTA" ), QVariant::LongLong ) );
    fields.append( QgsField( QStringLiteral( "POINTB" ), QVariant::LongLong ) );
    fields.append( QgsField( QStringLiteral( "POINTC" ), QVariant::LongLong ) );
  }
```
Same applies to Voronoi afaict. Thanks.

Reply:
It is intentional, this is behaviour of the original Python implementation.